### PR TITLE
Add Linux support

### PR DESCRIPTION
**Pull request by pepibumur**

### CODE REVIEW

**Comment by yonaskolb** [comment]
Interesting. Do we need to remove the auto generated equatable code? How is that affecting Linux? Would be shame to have to update those all by hand when making changes

**Comment by yonaskolb** [comment]
Ah, I see because of NSObject and @ObjC. We could have a custom Sourcery template then

**Comment by pepibumur** [comment]
I removed the autogenerated code and moved it to each individual object. I purposely removed the dependency with Sourcery to keep the project as lean as possible. Forgetting to run Sourcery and not adding new properties to the `isEqual` method result in the same thing so I'd rather not add the tool.

**Comment by pepibumur** [comment]
The package is building in Linux and tests are running. The last part missing is is adding all the tests to the `XCTestManifest` file. If all the tests run successfully, I think we are good to merge this PR.

**Comment by yonaskolb** [comment]
@pepibumur `swift test --generate-linuxmain`

**Comment by pepibumur** [comment]
> @pepibumur swift test --generate-linuxmain

🤯 Danke!

File: README.md [hunk 1]
```diff
@@ -108,6 +108,14 @@ object.
...
 [`swift-sh`]: https://github.com/mxcl/swift-sh
 
+## Docker
+
+If you want to test or build the package in a Linux environment, you can run the following command that uses the official Swift Docker image for that:
+
+```
+docker run --rm --volume "$(pwd):/package" --workdir "/package" swift:5.1.2 /bin/bash -c "swift package resolve && swift test --build-path ./.build/linux"
+```
+
```
Comment (kwridan) [inline README.md]: 👍 Having this documented makes it easy to test this out!

**Comment by pepibumur** [comment]
This PR is stale and I don't foresee working on this in the near future. I'll close it. If anyone is interested in accomplishing this one, feel free to open a new PR.

**Pull request closed without merging**